### DISCUSSION TITLE
ModuleNotFoundError: No module named 'cv2'

### DISCUSSION THREAD

**Discussion by bluekaiser07** [azuritecoin/OnnxDiffusersUI#122] [Q&A] [answered]
Before updating, it was still working.
I already tried reinstalling python 3.10.8, stable_diff, restarting pc, but still the same.
This is the error I'm having now. Any help on how to fix this?

(virtualenv) D:\stable_diff>python onnxUI.py --release-memory-after-generation --release-memory-on-change --cpu-textenc
Traceback (most recent call last):
  File "D:\stable_diff\onnxUI.py", line 7, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

These are my versions if it is of any help:
normal cmd
Python 3.10.8
git version 2.39.1.windows.1
pip 22.2.2

virtualenv cmd
Python 3.10.8
git version 2.39.1.windows.1
pip 23.0.1

I also updated the versions but still the same error.
normal cmd
Python 3.11.2
git version 2.40.0.windows.1
pip 22.3.1

virtualenv cmd
Python 3.11.2
git version 2.40.0.windows.1
pip 23.0.1

Additionally, I don't know why but I am also getting this errors when trying to update

WARNING: Skipping onnxruntime-directml as it is not installed.

ERROR: Could not find a version that satisfies the requirement onnxruntime-directml (from versions: none)
ERROR: No matching distribution found for onnxruntime-directml

**Comment by uchuusen** [comment, marked as answer]
I think that's because of the new controlnet update. The requirements.txt probably needs to be updated to reflect it. In the meantime, try installing opencv-python in your venv with this command:
`pip install opencv-python`
As for onnxruntime-directml, I don't know why that's not working for you.
Edit:
You may also need controlnet_aux as well:
`pip install controlnet_aux`

**Reply by bluekaiser07** [reply]
Installing opencv-python in virtualenv worked.
I don't know about directml, installed controlnet_aux but still not working. Then reinstalling everything somehow installed it, I don't have any idea, it is not installing before and I tried several times, is it because of the python version?
